pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -1026,6 +1044,155 @@ func registriesConfigIgnition(templateDir string, controllerConfig *mcfgv1.Contr
+			if err != nil {
+				return fmt.Errorf("error marshalling crio-default-container-runtime config ignition: %w", err)
+			}
+			mc.Spec.Config.Raw = rawRuntimeIgnition
+			// Create the crio-default-container-runtime MC
+			if err := retry.RetryOnConflict(updateBackoff, func() error {
+				_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Create(context.TODO(), mc, metav1.CreateOptions{})
+				return err
+			}); err != nil {
+				return fmt.Errorf("could not create MachineConfig for crio-default-container-runtime: %w", err)
+			}
+			klog.Infof("Applied default runtime MC %v on MachineConfigPool %v", managedKey, pool.Name)
+		} else {
+			klog.Infof("default_runtime is already set in existing MachineConfigs for pool %s, skipping MachineConfig creation", pool.Name)
+			continue
+		}
+	}
+
+	// Create the config map for crio-default-container-runtime so we know that the crio-default-container-runtime MC has been created
+	if defaultContainerRuntimeCM == nil {
+		defaultContainerRuntimeCM = &v1.ConfigMap{}
+	}
+
+	defaultContainerRuntimeCM.Name = configMapName
+	defaultContainerRuntimeCM.Namespace = ctrl.namespace
+	if _, err := ctrl.kubeClient.CoreV1().ConfigMaps(ctrl.namespace).Create(context.TODO(), defaultContainerRuntimeCM, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error creating %s config map: %w", configMapName, err)
+	}
+	return nil
+}
+
+func (ctrl *Controller) determineContainerRuntimeForFinalRuntimeConfig(pool *mcfgv1.MachineConfigPool) (mcfgv1.ContainerRuntimeDefaultRuntime, error) {
+	// Retrieve all ContainerRuntimeConfig objects for the pool
+	ctrcfgList, err := getContainerRuntimeConfigsForPool(ctrl.client, pool)
+	if err != nil {
+		return "", fmt.Errorf("error retrieving container runtime configs for pool %s: %w", pool.Name, err)
+	}
+
+	// Determine the final runtime configuration
+	var currentDefaultRuntime mcfgv1.ContainerRuntimeDefaultRuntime
+	var machineConfigs []*mcfgv1.MachineConfig
+
+	for _, ctrcfg := range ctrcfgList {
+		finalizers := ctrcfg.GetFinalizers()
+		for _, finalizer := range finalizers {
+			mcName := finalizer
+			mc, err := ctrl.client.MachineconfigurationV1().MachineConfigs().Get(context.TODO(), mcName, metav1.GetOptions{})
+			if err != nil {
+				return "", fmt.Errorf("error getting MachineConfig %s: %w", mcName, err)
+			}
+			machineConfigs = append(machineConfigs, mc)
+		}
+	}
+
+	// Sort MachineConfigs in alphanumeric order
+	sort.Slice(machineConfigs, func(i, j int) bool {
+		return machineConfigs[i].Name < machineConfigs[j].Name
+	})
+
+	// Determine the final runtime configuration based on the last-applied MachineConfig
+	for i := len(machineConfigs) - 1; i >= 0; i-- {
+		mc := machineConfigs[i]
+		if mc.Spec.Config.Raw != nil {
+			ignConfig := ctrlcommon.NewIgnConfig()
+			if err := json.Unmarshal(mc.Spec.Config.Raw, &ignConfig); err != nil {
+				return "", fmt.Errorf("error unmarshalling Ignition config: %w", err)
+			}
```

Comment:
we can reuse ParseAndConvertConfig
```suggestion
			ignConfig, err := ctrlcommon.ParseAndConvertConfig(mc.Spec.Config.Raw)
			if err != nil {
				return "", fmt.Errorf("error unmarshalling Ignition config: %w", err)
			}
```